doc/source/ray-air/examples/analyze_tuning_results.ipynb

```diff
@@ -0,0 +1,918 @@
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Trial XGBoostTrainer_4544b_00007 completed. Last result: train-mlogloss=0.607805,train-merror=0.220687,should_checkpoint=True\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "2022-05-09 17:29:41,405\tINFO tune.py:753 -- Total run time: 246.90 seconds (245.40 seconds for the tuning loop).\n"
+     ]
+    }
+   ],
+   "source": [
+    "results = tuner.fit()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2c7c444e",
+   "metadata": {},
+   "source": [
+    "Now that we obtained the results, we can analyze them. For instance, we can fetch the best observed result according to the configured `metric` and `mode` and print it:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "4f4e5187",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Best result error rate 0.197125\n"
+     ]
+    }
+   ],
+   "source": [
+    "# This will fetch the best result according to the `metric` and `mode` specified\n",
+    "# in the `TuneConfig` above:\n",
+    "\n",
+    "best_result = results.get_best_result()\n",
+    "\n",
+    "print(\"Best result error rate\", best_result.metrics[\"train-merror\"])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "53d71b08",
+   "metadata": {},
+   "source": [
+    "For more sophisticated analysis, we can get a pandas dataframe with all trial results:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "50e76e91",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Index(['train-mlogloss', 'train-merror', 'time_this_iter_s',\n",
+      "       'should_checkpoint', 'done', 'timesteps_total', 'episodes_total',\n",
+      "       'training_iteration', 'trial_id', 'experiment_id', 'date', 'timestamp',\n",
+      "       'time_total_s', 'pid', 'hostname', 'node_ip', 'time_since_restore',\n",
+      "       'timesteps_since_restore', 'iterations_since_restore', 'warmup_time',\n",
+      "       'config/params', 'config/preprocessor', 'logdir'],\n",
+      "      dtype='object')\n"
+     ]
+    }
+   ],
+   "source": [
+    "df = results.get_dataframe()\n",
+    "print(df.columns)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b8a05668",
+   "metadata": {},
+   "source": [
+    "As an example, let's group the results per preprocessor and fetch the minimal obtained values:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "id": "94b017b4",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Preprocessor 1 minimum log loss: 0.559286\n",
+      "Preprocessor 2 minimum log loss: 0.559286\n",
+      "Preprocessor 1 minimum error: 0.197125\n",
+      "Preprocessor 2 minimum error: 0.197125\n"
```

Comment:
Seems like choice of preprocessor doesn't make a difference?

Reply:
Should we add a sentence explaining the takeaway (so as a user I know how to interpret these results)?

Reply:
Yeah actually I'll have to either leave out preprocessing here or find something else that actually makes sense. Decision trees don't care about monotonic transformations...

Reply:
encoding categorical columns, maybe?

Reply:
The dataset is pretty much perfect in terms of preprocessing. Let's do dataset and preprocessor tuning somewhere else.
We should probably also focus on showing just one thing per example, and in this case this would be the analysis of the results dataframe.